tests/unit/tracerTest.cpp

```diff
@@ -0,0 +1,63 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+
+#include <tilck/common/basic_defs.h>
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+#include "mocking.h"
+
+using namespace testing;
+
+extern "C" {
+#include <tilck/common/string_util.h>
+#include <tilck/common/printk.h>
+
+#include <tilck/kernel/user.h>
+#include <tilck/mods/tracing.h>
+
+bool save_param_buffer(void *, long, char *, size_t);
+bool dump_param_buffer(ulong, char *, long, long, char *, size_t);
+}
+
+class MockingTracer : public KernelSingleton {
+public:
+
+   MOCK_METHOD(int, copy_str_from_user, (void *, const void *, size_t, size_t *), (override));
+   MOCK_METHOD(int, copy_from_user, (void *, const void *, size_t), (override));
+};
+
+TEST(tracer_test, save_param_buffer)
+{
+   MockingTracer mock;
+
+   long data_sz = -1;
+   void * data = (void *)"test";
```

Comment:
Style: no space between `*` and `data`.  Note: the cast for "test" is not really required in C.

Reply:
Without the cast, the following error is thrown:
```shell
error: invalid conversion from ‘const void*’ to ‘void*’ [-fpermissive]
   32 |    void *data = "test";
      |                 ^~~~~~
      |                 |
      |                 const void*

```